### PR TITLE
KAFKA-9309: Add the ability to translate Message classes to and from JSON

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/protocol/Message.java [hunk 1]
```diff
@@ -96,6 +98,42 @@
...
+     * Note that we expect to see NullNode objects created for null entries.
+     * Therefore, please configure your Jackson ObjectMapper with
+     * setSerializationInclusion({@link JsonInclude.Include#ALWAYS}).
+     * Other settings may silently omit the nulls, which is not the
+     * semantic that Kafka RPC uses.  (Including a field and setting it to
+     * null is different than not including the field.)
+     *
+     * @param node          The source node.
+     * @param version       The version to use.
+     */
```
Comment (rondagostino) [inline clients/src/main/java/org/apache/kafka/common/protocol/Message.java:127]: Needs `@throws {@see org.apache.kafka.common.errors.UnsupportedVersionException}` (so does `fromStruct()`, BTW -- it is missing).

Comment (cmccabe) [reply clients/src/main/java/org/apache/kafka/common/protocol/Message.java:127]: added

**Comment by cmccabe** [comment]
retest this please

**Review by mumrah — COMMENTED** [review]
Looks good overall. Would it be worth adding a unit test to convert from JSON to a Message and then back?

**Comment by cmccabe** [comment]
Test failure is `org.apache.kafka.streams.integration.QueryableStateIntegrationTest.concurrentAccesses`, which is not related (it's a streams test)

**Review by mumrah — APPROVED** [review]
👍

**Pull request merged**